beacon-chain/state/stateutil/trie_helpers.go

```diff
@@ -61,25 +63,41 @@ func ReturnTrieLayerVariable(elements [][32]byte, length uint64) [][]*[32]byte {
 	layers[0] = transformedLeaves
 	buffer := bytes.NewBuffer([]byte{})
 	buffer.Grow(64)
+
 	for i := 0; i < int(depth); i++ {
-		oddNodeLength := len(layers[i])%2 == 1
-		if oddNodeLength {
-			zerohash := trie.ZeroHashes[i]
-			layers[i] = append(layers[i], &zerohash)
-		}
-		updatedValues := make([]*[32]byte, 0, len(layers[i])/2)
-		for j := 0; j < len(layers[i]); j += 2 {
-			buffer.Write(layers[i][j][:])
-			buffer.Write(layers[i][j+1][:])
-			concat := hasher(buffer.Bytes())
-			updatedValues = append(updatedValues, &concat)
-			buffer.Reset()
-		}
-		// remove zerohash node from tree
-		if oddNodeLength {
-			layers[i] = layers[i][:len(layers[i])-1]
+		layerLen := len(layers[i])
+		oddNodeLength := layerLen%2 == 1
+		if features.Get().EnableVectorizedHTR {
+			if oddNodeLength {
+				zerohash := trie.ZeroHashes[i]
+				elements = append(elements, zerohash)
+				layerLen++
+			}
+			length := math.Max(uint64(layerLen/2), 1)
```

Comment:
Why the Max here? if `layerLen > 0` then the minimum is 1 anyway.

Reply:
I remember it constantly crashing without this line, but will double check if it can be removed.

Reply:
if it crashes then we have a bad bug sending length zero fields (or there's an off by one and we end the loop with a length zero field)